core/src/main/java/smile/classification/FLD.java

```diff
@@ -273,32 +278,51 @@ public FLD(double[][] x, int[] y, int L, double tol) {
         }
 
         T.setSymmetric(true);
-        EVD eigen = T.eigen();
+        EVD eigen = T.eigen();  
         
         tol = tol * tol;
         double[] s = eigen.getEigenValues();
         for (int i = 0; i < s.length; i++) {
             if (s[i] < tol) {
                 throw new IllegalArgumentException("The covariance matrix is close to singular.");
             }
-
-            s[i] = 1.0 / s[i];
```

Comment:
why remove this?

Reply:
not quite clear about the usage of this line